### PR TITLE
[Hotfix][core] Fix spark engine parallelism parameter does not working

### LINKED ISSUE #2931: [Bug] [Connector-V2-Spark] 'source.parallelism' parameter does not work in spark engine
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

Now the parameter of setting parallelism for source connector does not work in spark engine.

### SeaTunnel Version

dev

### SeaTunnel Config

```conf
none
```

### Running Command

```shell
none
```

### Error Exception

```log
none
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

fix #2931 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
@ashulin Please review.

File: seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/SourceExecuteProcessor.java [hunk 1]
```diff
@@ -57,6 +57,7 @@ public List<Dataset<Row>> execute(List<Dataset<Row>> upstreamDataStreams) {
             Dataset<Row> dataset = sparkEnvironment.getSparkSession()
                 .read()
                 .format(SeaTunnelSource.class.getSimpleName())
+                .option(Constants.PARALLELISM, sparkEnvironment.getSparkConf().get(Constants.PARALLELISM))
```
Comment (ashulin) [inline seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/SourceExecuteProcessor.java]: Can parallelism be set for a single plugin? If the plugin don‘t set the parallelism, get the global parallelism.

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/Constants.java [hunk 2]
```diff
@@ -31,7 +31,7 @@ public final class Constants {
 
     public static final String SOURCE_SERIALIZATION = "source.serialization";
 
-    public static final String SOURCE_PARALLELISM = "source.parallelism";
+    public static final String SOURCE_PARALLELISM = "parallelism";
```
Comment (ashulin) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/Constants.java:34]: marke as @Deprecated

Comment (TyrantLucifer) [reply seatunnel-common/src/main/java/org/apache/seatunnel/common/Constants.java:34]: Resolved.

File: seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/SourceExecuteProcessor.java (hunk 1 above)
Comment (TyrantLucifer) [reply seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/execution/SourceExecuteProcessor.java]: Resolved.

**Review by ashulin — APPROVED** [review]

**Review by hailin0 — COMMENTED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]
+1

**Pull request merged**